### PR TITLE
HCLIND-54 Please check and update Azure instance types - v2

### PR DESCRIPTION
**Pull request by Rangelmv23**
### Description

We've had a complaint from the customer that certain Azure instance types are not listed inside this Policy Template: https://app.flexera.com/orgs/27018/policy/catalog?templateId=5d161b33e074030099620f63

### Issues Resolved

Updated azure instance_types.json file with latest VM's from azure official documentation: https://learn.microsoft.com/en-us/azure/virtual-machines/sizes (removed superseded recommendations for constrained vCPU sizes).

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: data/azure/instance_types.json [hunk 1]
```diff
@@ -335,111 +335,111 @@
...
-  "Standard_DS11-1_v2": {"up": null, "down": null, "superseded": {"regular": "Standard_DS11_v2"}},
-  "Standard_DS12-1_v2": {"up": "Standard_DS12-2_v2", "down": null, "superseded": {"regular": "Standard_DS12_v2"}},
-  "Standard_DS12-2_v2": {"up": null, "down": "Standard_DS12-1_v2", "superseded": {"regular": "Standard_DS12_v2"}},
-  "Standard_DS13-2_v2": {"up": "Standard_DS13-4_v2", "down": null, "superseded": {"regular": "Standard_DS13_v2"}},
-  "Standard_DS13-4_v2": {"up": null, "down": "Standard_DS13-2_v2", "superseded": {"regular": "Standard_DS13_v2"}},
-  "Standard_DS14-4_v2": {"up": "Standard_DS14-8_v2", "down": null, "superseded": {"regular": "Standard_DS14_v2"}},
-  "Standard_DS14-8_v2": {"up": null, "down": "Standard_DS14-4_v2", "superseded": {"regular": "Standard_DS14_v2"}},
-  "Standard_M416-208s_v2": {"up": null, "down": null, "superseded": {"regular": "Standard_M416s_v2"}},
-  "Standard_M416-208ms_v2": {"up": null, "down": null, "superseded": {"regular": "Standard_M416ms_v2"}},
+  "Standard_M8-2ms": {"up": "Standard_M8-4ms", "down": null, "superseded": null},
```
Comment (bryankaraffa) [inline data/azure/instance_types.json:338]: I am not familiar enough with Azure Instance types to know this off-hand.  Based on "removed superseded recommendations for constrained vCPU sizes" comment in PR description, seems like this _may_ be correct

Comment (Rangelmv23) [reply data/azure/instance_types.json:338]: @bryankaraffa Do you think that this PR could be approved? Or Is there anyone who could help us to check if this is right?

File: data/azure/instance_types.json [hunk 2]
```diff
@@ -335,111 +335,111 @@
...
+  "Standard_M128-32ms": {"up": "Standard_M128-64ms", "down": null, "superseded": null},
+  "Standard_M128-64ms": {"up": null, "down": "Standard_M128-32ms", "superseded": null},
+  "Standard_E4-2s_v3": {"up": null, "down": null, "superseded": null},
+  "Standard_E8-2s_v3": {"up": "Standard_E8-4s_v3", "down": null, "superseded": null},
+  "Standard_E8-4s_v3": {"up": null, "down": "Standard_E8-2s_v3", "superseded": null},
+  "Standard_E16-4s_v3": {"up": "Standard_E16-8s_v3", "down": null, "superseded": null},
+  "Standard_E16-8s_v3": {"up": null, "down": "Standard_E16-4s_v3", "superseded": null},
+  "Standard_E32-8s_v3": {"up": "Standard_E32-16s_v3", "down": null, "superseded": null},
+  "Standard_E32-16s_v3": {"up": null, "down": "Standard_E32-8s_v3", "superseded": null},
+  "Standard_E64-16s_v3": {"up": "Standard_E64-32s_v3", "down": null, "superseded": {null},
```
Comment (nia-vf1) [inline data/azure/instance_types.json:355]: Got an extra squiggly bracket here 
```suggestion
  "Standard_E64-16s_v3": {"up": "Standard_E64-32s_v3", "down": null, "superseded": null},
```

File: data/azure/instance_types.json (hunk 1 above)
Comment (nia-vf1) [reply data/azure/instance_types.json:338]: Agree, think this looks good based on https://learn.microsoft.com/en-us/azure/virtual-machines/constrained-vcpu

File: data/azure/instance_types.json (hunk 2 above)
Comment (nia-vf1) [reply data/azure/instance_types.json:355]: Don't ask how long it took me to find this 😆

Comment (Rangelmv23) [reply data/azure/instance_types.json]: Didn't notice. Thank you so much for that!

**Review by nia-vf1 — APPROVED** [review]
LGTM

**Review by degarajesh — APPROVED** [review]
LGTM

**Pull request merged**